src/main/navigation/navigation_fixedwing.c

```diff
@@ -463,7 +463,23 @@ int16_t applyFixedWingMinSpeedController(timeUs_t currentTimeUs)
 
 int16_t fixedWingPitchToThrottleCorrection(int16_t pitch)
 {
-    return DECIDEGREES_TO_DEGREES(pitch) * navConfig()->fw.pitch_to_throttle;
+    // Calculate base throttle correction from pitch moving average
+    const int16_t movingAverageCycles = navConfig()->fw.pitch_to_throttle_smooth;
+    static int16_t averagePitch = 0;
+
+    averagePitch = (averagePitch * movingAverageCycles + pitch - averagePitch) / movingAverageCycles;
+    const int16_t baseThrottleCorrection = DECIDEGREES_TO_DEGREES(averagePitch) * navConfig()->fw.pitch_to_throttle;
+
+    // Calculate final throttle correction
+    if (pitch > (averagePitch + navConfig()->fw.pitch_to_throttle_thresh)) {
+        return baseThrottleCorrection + DECIDEGREES_TO_DEGREES(pitch - averagePitch - navConfig()->fw.pitch_to_throttle_thresh) * navConfig()->fw.pitch_to_throttle;
```

Comment:
This essentially calculates the throttle correction based on  `pitch - fw.pitch_to_throttle_thresh`, essentially clipping the response outside of the deadband.

Wouldn't it make sense to calculate the out of deadband throttle correction based on `pitch` itself, allowing us to support rapid change in pitch with equally rapid change in throttle? Then, once the `averagePitch` catches up, the response will be smoothed again.

WDYT?

Reply:
Since `fw.pitch_to_throttle_thresh` is fixed, isn't this equivalent?

Reply:
No, `pitch * fw.pitch_to_throttle` > `(pitch - fw.pitch_to_throttle_thresh) * fw.pitch_to_throttle`.

Your PR not only smooths the response, it also clips it considerably. Basically, the maximum possible throttle correction (`min/max_pitch * fw.pitch_to_throttle`) with this PR will never be reached initially, only after `averagePitch` catches up. 

With a large enough averaging delay and rapid enough pitch maneuver it may cause a stall in a configuration that would fly fine without this PR.

Reply:
Hmm, I see what you mean but as long as the moving average is calculated over a not too long window I highly doubt there will be problems with this PR. But I agree that a filter on the pitch angle may be a cleaner approach.

This is not my PR, btw. I just requested the feature and got involved from there. I'll look into this PT1 filter and see if I can come up with something, but perhaps @Airwide wants to do that first.

Reply:
@digitalentity Thanks for you feedback The calculation as it is right now, will make a _partial_ throttle compensation for pitch deviations outside the deadband, until the moving average catches up.

Reply:
Hi, I did test the behaviour of threshold averaging by @Airwide. It works quite fine. Helps a lot with constantly waving rpms on my BushMule.
I noticed that when you switch off averaging (setting smoothness to 1)  threshold has no effect.

Regards,
Jacek

Reply:
@Airwide  I don't like the **partial** part of this PR. I like the deadband and smoothing the response when it's not necessary, but please change the logic to allow the throttle compensation to use the full extent of the throttle compensation in the extremes.

Reply:
@digitalentity Changed to full throttle compensation outside of deadband.